enhancements/hypershift/trust-chain-separation.md

```diff
@@ -0,0 +1,203 @@
+  - ibihim
+  - deads2k
+  - alvaroaleman
+  - csrwng
+approvers:
+  - deads2k
+  - alvaroaleman
+  - csrwng
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AUTH-311
+see-also:
+replaces:
+superseded-by:
+---
+
+# Trust Chain Separation
+
+## Summary
+
+At the time of writing the Hypershift is generating most of its certificates
+by using a single trust-all root CA. This enhancement describes how and why
+such a bundle should be split in order to assure workload portability between
+OpenShift-based products.
+
+## Motivation
+
+The OpenShift certificate trust domain separation has been serving well
+to prevent bugs of extended trust for client certificate authentication,
+and also to prevent implicit trust to servers that the client should not
+attempt connecting to in the first place.
+
+Having a single signer for most certificates is suboptimal as it opens
+window for bugs with unexpected trusts where a server/client grabs the
+global CA bundle and misuses it as this likely "just works" for them.
+It is therefore possible to build a number of solutions which would be
+considered "stable". As that effectively makes the CA bundle an "API",
+it is later hard to split the trust to be more fine-grained as chipping
+away threads of trust breaks the solutions which now rely on the global
+trust bundle.
+
+The intention is to perform at least a minimal split of the client certificate
+signers from the root CA bundle in order to prevent unwanted application
+authentications and to assure portability of user workload among OpenShift
+platform flavors.
+
+### User Stories
+
+1. I would like to be able to run my workloads on any OpenShift-like platform.
+My workloads allow components to authenticate using client certificates based
+on a trust bundle that I am able to retrieve from the cluster.
+
+2. I don't want my users to have access to any CA bundle that would allow them
+to trust a random certificate from the cluster for client certificate authentication.
+
+### Goals
+
+1. Separate client trust from the global root CA so that client
+certificates are being minted by CAs outside of the global trust
+and these CAs are separate so that they create logical trust domains.
+
+2. Make sure that it is possible to supply the following certificates for
+the kube-apiserver:
+    - client certificate to trust for a worst-case break-glass scenario
+    - CA and client certificate to trust for API aggregation
+    - serving certificate for a given hostname to enable SNI for external domains
+
+### Non-Goals
+
+1. Separate the root CA completely so that the trust chains resemble those
```

Comment:
to clarify, the only reason this isn't a goal is because of time constraints.  otherwise, I suspect we'd want to match more or less exactly.  OCP has aged pretty well.